### PR TITLE
test: Use the new pool abi for tests

### PR DESCRIPTION
**Pull request by RabbitDoge**
The olf ABI is currently still compatible with the new contract.
Execpt for the tests, where we need to fetch the staking token.

[ ] Before opening a pull request, please read the [contributing guidelines](https://github.com/pancakeswap/pancake-frontend/blob/master/CONTRIBUTING.md) first
[ ] If your PR is work in progress, open it as `draft`
[ ] Before requesting a review, all the checks need to pass
[ ] Explain what your PR does

### CODE REVIEW

File: src/utils/contractHelpers.ts [hunk 1]
```diff
@@ -71,6 +72,10 @@ export const getSouschefContract = (id: number, web3?: Web3) => {
   const abi = config.poolCategory === PoolCategory.BINANCE ? sousChefBnb : sousChef
   return getContract(abi, getAddress(config.contractAddress), web3)
 }
+export const getSouschefV2Contract = (id: number, web3?: Web3) => {
```
Comment (ChefHutch) [inline src/utils/contractHelpers.ts:75]: I don't really like littering the codebase with comments... but do you think it's worth adding something here explaining that this contract is only needed for the pools tests when calling the `stakedToken` method?

File: src/__tests__/config/pools.test.ts [hunk 2]
```diff
@@ -29,8 +29,15 @@ describe('Config pools', () => {
   it.each(poolsToTest.filter((pool) => pool.stakingToken.symbol !== 'BNB'))(
     'Pool %p has the correct staking token',
     async (pool) => {
-      const contract = getSouschefContract(pool.sousId)
-      const stakingTokenAddress = await contract.methods.syrup().call()
+      let stakingTokenAddress = null
+      try {
+        const contract = getSouschefV2Contract(pool.sousId)
+        stakingTokenAddress = await contract.methods.stakedToken().call()
+      } catch (error) {
```
Comment (ChefHutch) [inline src/__tests__/config/pools.test.ts:36]: Is the `contract.methods.stakedToken()` call compatible (in theory) with _all_ pools, or just the new ones? 

Just wondering whether it's going to attempt this call, error, and then attempt the `syrup` call for every pool other than the new ones?

**Review by ChefHutch — APPROVED** [review]
Couple of Qs but lgtm

**Review by Chef-Chungus — APPROVED** [review]

File: src/__tests__/config/pools.test.ts (hunk 2 above)
Comment (RabbitDoge) [reply src/__tests__/config/pools.test.ts:36]: syrup has been replaced by stakedToken in the new contract, it's basically the only breaking change on the abi, and it was used only in the tests. I didn't find a cleaner solution , but if you have any suggestion you can propose a different implementation

**Pull request merged**